src/systemctl/systemctl-logind.c

```diff
@@ -204,6 +204,10 @@ int logind_check_inhibitors(enum action a) {
         if (r < 0)
                 return bus_log_parse_error(r);
 
+        /* root respects inhibitors since v257 but keeps ignoring sessions by default */
+        if (arg_check_inhibitors < 0 && c == 0 && geteuid() == 0)
+                return 0;
```

Comment:
hmm, what precisely does this undo? i cannot recognize what this restores from the original patch

Reply:
The commit message attempts to explain it:

"The change was supposed to be about respecting inhibitors, but
it was extended to also error out when there are active user
sessions, which was not intentional. Restore the previous
behaviour."

If you check the linked commit in the message, it refers to specifically this line:

https://github.com/systemd/systemd/commit/804874d26ac73e0af07c4c5d7165c95372f03f6d#diff-9e3ed6eaaf2934d1336b1d346d7f291e2ab899dbe882f9281c241ec853d2297cL148

ie, before if you called systemctl as root or root-equivalent it skipped checking both inhibitors _and_ for active sessions. The original change removed that shortcut, but failed to retain it when there are no active inhibitors but there are active sessions. This change restores the behaviour, if there are no active inhibitors and you are root, the check for sessions is skipped once again.